app/policies/course_policy.rb

```diff
@@ -6,7 +6,7 @@ class CoursePolicy < ApplicationPolicy
   alias_rule :clear_role_switch_session?, to: :role_is_switched?
 
   def show?
-    role.admin?
+    !role.course.is_hidden? && !role.hidden?
```

Comment:
Admins can see hidden courses. Courses should only be hidden to students.

Reply:
Right, forgot to include admins as always able to see even if the course is hidden. Updated this to:

`role.admin? || (!role.course.is_hidden? && !role.hidden?)`

Because students should still be able to access the show view if the course is not hidden and their role isn't hidden

Reply:
students don't access this route. The landing page for students for a course is `course/:course_id/assignments` not `course/:course_id`

Reply:
Right, that's where I was missing a piece. fair enough, reverted.